src/doc/rustc/src/platform-support/nuttx.md

```diff
@@ -0,0 +1,60 @@
+# `*-nuttx-elf`
+
+**Tier: 3**
+
+Targets for the [Apache NuttX](https://github.com/apache/nuttx).
+
+Apache NuttX is a real-time operating system (RTOS) with an emphasis on standards compliance and small footprint. It is scalable from 8-bit to 64-bit microcontroller environments. The primary governing standards in NuttX are POSIX and ANSI standards.
+
+NuttX adopts additional standard APIs from Unix and other common RTOSs, such as VxWorks. These APIs are used for functionality not available under the POSIX and ANSI standards. However, some APIs, like fork(), are not appropriate for deeply-embedded environments and are not implemented in NuttX.
+
+For brevity, many parts of the documentation will refer to Apache NuttX as simply NuttX.
+
+## Target maintainers
+
+- Qi Huang [@no1wudi](https://github.com/no1wudi)
+
+## Requirements
+
+The target name follow this format: `ARCH[-VENDOR]-nuttx-ABI`, where `ARCH` is the target architecture, `VENDOR` is the vendor name, and `ABI` is the ABI used.
```

Comment:
Why is the vendor part left off for thumb targets?

Reply:
These thumb targets are derived from bare-metal thumb targets like `thumbv7m-none-eabi`, should we append vendor part to them as riscv does (`thumbv7m-unknown-nuttx-eabi`)?

Reply:
I see. I think this is fine as long as it is consistent with the other thumb targets.